src/cpp/editor/hero_edit.cpp

```diff
@@ -1,17 +1,39 @@
-//#include "base.h"
-
-#include "adventure/map.h"
+#include "hero_edit.h"
 
+#include "artifacts.h"
 #include "editor.h"
 #include "gui/msg.h"
 
 extern HeroExtra gEditedHeroExtra;
 
-extern int __fastcall EditHeroHandler_orig(tag_message&);
-extern void __stdcall FillInHeroEdit(HeroExtra*);
-
 #define NUM_HERO_PORTRAITS 72
 
+namespace {
+  bool fillInArtifacts = true;
+}
+
+void __stdcall FillInHeroEdit(HeroExtra *extra) {
+  const int ARTIFACT_SLOT_1 = 308;
+
+  if (fillInArtifacts) {
+    tag_message artifactEvent;
+    artifactEvent.eventCode = INPUT_GUI_MESSAGE_CODE;
+    artifactEvent.xCoordOrKeycode = GUI_MESSAGE_DROPLIST_ADD;
+    for (int i = MAX_EXPANSION_ARTIFACT + 1; i < NUM_SUPPORTED_ARTIFACTS; ++i) {
+      if (gArtifactNames[i]) {
+        for (int j = 0; j < 3; ++j) {
+          artifactEvent.payload = gArtifactNames[i];
+          artifactEvent.yCoordOrFieldID = j + ARTIFACT_SLOT_1;
```

Comment:
This is the only thing I'm insisting on. Create a GUIAddDroplistRow function and use it instead, in accordance with http://projectironfist.pbworks.com/w/page/115903900/Working%20with%20the%20GUI